lib/nerves_hub_link/extensions/logging/logging.ex

```diff
@@ -0,0 +1,60 @@
+# SPDX-FileCopyrightText: 2025 Josh Kalderimis
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+defmodule NervesHubLink.Extensions.Logging do
+  @moduledoc """
+  The Logging Extension.
+
+  Send logs to NervesHub for easy debugging.
+
+  This extension is disabled by default while in Early Release.
+  """
+  use NervesHubLink.Extensions, name: "logging", version: "0.0.1"
+
+  require Logger
+
+  @doc """
+  Request a log payload be sent asynchronously.
+
+  ## Examples
+
+      iex> NervesHubLink.Extensions.Logging.send_logs(:info, "hello, it's me", %{sensor: "door lock"})
+      :ok
+
+  """
+  @spec send_log_line(atom(), String.t(), map()) :: :ok
+  def send_log_line(level, message, meta) do
+    GenServer.cast(__MODULE__, {:send_logs, %{level: level, message: message, meta: meta}})
+  end
+
+  @impl GenServer
+  def init(_opts) do
+    _ =
+      :logger.add_handler(
+        :nerves_hub_link_logger_extension_handler,
+        NervesHubLink.Extensions.Logging.LoggerHandler,
+        %{}
+      )
+
+    {:ok, %{}}
+  end
+
+  @impl NervesHubLink.Extensions
+  def handle_event(_, _msg, state) do
+    {:noreply, state}
+  end
+
+  @impl GenServer
+  def handle_cast({:send_logs, log_payload}, state) do
+    _ = send_logs(log_payload, state)
+    {:noreply, state}
+  end
+
+  defp send_logs(log_payload, state) do
+    case push("send", log_payload) do
```

Comment:
I think we should have some sort of rate limiting on the client side as well. Otherwise we're going to bloat the connection which also controls updates

Reply:
Maybe would be worthwhile to batch logs for ~500ms or something. Means we get natural batches on the insert side as well. Not exactly rate limiting but does improve the rate potentially.

More rate-limity: If a device is spewing logs it would make sense to shed some of them (it is probably looping/spewing the same thing continuously) and just include a log line about having shed logs?